### PR TITLE
{Pylint} Fix implicit-str-concat

### PR DESCRIPTION
**Pull request by bebound**
Fix this with `ruff check . --select ISC001  --fix`

superfluous-parens was disabled in https://github.com/Azure/azure-cli/pull/26685

Ref:
https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/implicit-str-concat.html
https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/cdn/custom/custom_rule_util.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def create_action(action_name, cache_behavior=None, cache_duration=None, header_
         action = {
             "cache_key_query_string": {
                 "parameters": {
-                    "type_name" "DeliveryRuleCacheKeyQueryStringBehaviorActionParameters"
+                    "type_nameDeliveryRuleCacheKeyQueryStringBehaviorActionParameters"
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/cdn/custom/custom_rule_util.py:295]: This original code seems to be wrong.

```suggestion
                    "type_name": "DeliveryRuleCacheKeyQueryStringBehaviorActionParameters"
```

File: src/azure-cli/azure/cli/command_modules/relay/tests/latest/test_relay_commands.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def test_relay_hyco(self, resource_group):
             'loc': 'westus2',
             'rg': resource_group,
             'namespacename': self.create_random_name(prefix='relay-nscli', length=20),
-            'tags': 'tag1=value1' 'tag2=value2',
+            'tags': 'tag1=value1tag2=value2',
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/relay/tests/latest/test_relay_commands.py:105]:
```suggestion
            'tags': 'tag1=value1 tag2=value2',
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -1136,7 +1136,7 @@ def load_arguments(self, _):
...
             c.argument('nics', nargs='+', help='Names or IDs of existing NICs to attach to the VM. The first NIC will be designated as primary. If omitted, a new NIC will be created. If an existing NIC is specified, do not specify subnet, VNet, public IP or NSG.')
             c.argument('private_ip_address', help='Static private IP address (e.g. 10.0.0.5).')
-            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" for None (\'""\' in Azure CLI using PowerShell or --% operator). For Azure CLI using powershell core edition 7.3.4, specify '' or "" (--public-ip-address '' or --public-ip-address "")')
+            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" for None (\'""\' in Azure CLI using PowerShell or --% operator). For Azure CLI using powershell core edition 7.3.4, specify  or "" (--public-ip-address  or --public-ip-address "")')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1139]:
```suggestion
            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" for None (\'""\' in Azure CLI using PowerShell or --% operator). For Azure CLI using powershell core edition 7.3.4, specify or "" (--public-ip-address  or --public-ip-address "")')
```

File: src/azure-cli/azure/cli/command_modules/cdn/custom/custom_rule_util.py (hunk 1 above)
Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/cdn/custom/custom_rule_util.py:295]: Also missing `,` at the end of file.

Previously, `type_nameDeliveryRuleCacheKeyQueryStringBehaviorActionParametersquery_string_behavior` is the first key.

File: src/azure-cli/azure/cli/command_modules/relay/tests/latest/test_relay_commands.py [hunk 4]
```diff
@@ -185,7 +185,7 @@ def test_relay_wcfrelay(self, resource_group):
             'rg': resource_group,
             'relaytype': 'Http',
             'namespacename': self.create_random_name(prefix='relay-nscli', length=20),
-            'tags': 'tag1=value1' 'tag2=value2',
+            'tags': 'tag1=value1tag2=value2',
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/relay/tests/latest/test_relay_commands.py:188]:
```suggestion
            'tags': 'tag1=value1 tag2=value2',
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 3 above)
Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:1139]: The help message is not correct. In powershell, the empty string shoule be represented as  `'""'`.

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py [hunk 5]
```diff
@@ -91,7 +91,7 @@ def load_arguments(self, _):
             c.argument('sql_admin_login_password', options_list=['--sql-admin-login-password', '-p'],
                        help='The sql administrator login password.')
             c.argument('tags', arg_type=tags_type)
-            c.argument('allowed_aad_tenant_ids', options_list=['--allowed-tenant-ids'], nargs='+', help="The approved Azure AD tenants which outbound data traffic allowed to. The Azure AD tenant of the current user will be included by default. Use ""(\'""\' in PowerShell) to disable all allowed tenant ids.")
+            c.argument('allowed_aad_tenant_ids', options_list=['--allowed-tenant-ids'], nargs='+', help="The approved Azure AD tenants which outbound data traffic allowed to. The Azure AD tenant of the current user will be included by default. Use (\'\' in PowerShell) to disable all allowed tenant ids.")
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:94]:
```suggestion
            c.argument('allowed_aad_tenant_ids', options_list=['--allowed-tenant-ids'], nargs='+', help="The approved Azure AD tenants which outbound data traffic allowed to. The Azure AD tenant of the current user will be included by default. Use \"\" or '' ('\"\"' in PowerShell) to disable all allowed tenant ids.")
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 6]
```diff
@@ -438,7 +438,7 @@ def load_arguments(self, _):
         c.argument('dapr_enable_api_logging', options_list=['--dapr-enable-api-logging', '--dal'], help="Enable/Disable API logging for the Dapr sidecar.", arg_type=get_three_state_flag(return_label=True))
         c.argument('workload_profile_name', help="The name of the workload profile to run the app on.", is_preview=True)
         c.argument('cpu', type=float, help="Required CPU in cores from 0.5 to 2.0.", is_preview=True)
-        c.argument('memory', help="Required momory from 1.0 to 4.0 ending with ""Gi"" e.g. 1.0Gi, ", is_preview=True)
+        c.argument('memory', help="Required momory from 1.0 to 4.0 ending with Gi e.g. 1.0Gi, ", is_preview=True)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:441]: I guess the author wants to quote `Gi`:

```suggestion
        c.argument('memory', help='Required momory from 1.0 to 4.0 ending with "Gi" e.g. 1.0Gi, ', is_preview=True)
```

File: src/azure-cli/azure/cli/command_modules/policyinsights/_params.py [hunk 7]
```diff
@@ -42,17 +42,17 @@ def load_arguments(self, _):
                 'namespace',
                 completer=get_providers_completion_list,
                 arg_group='Resource ID',
-                help='Provider namespace (Ex: ''Microsoft.Provider'').')
+                help='Provider namespace (Ex: Microsoft.Provider).')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/policyinsights/_params.py:45]:
```suggestion
                help="Provider namespace (Ex: 'Microsoft.Provider')."')
```

Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/policyinsights/_params.py:45]: I think the author actually wants to quote `Microsoft.Provider`, but used the PowerShell syntax:

```
> echo "a""b"
a"b
> echo 'a''b'
a'b
```

```suggestion
                help="Provider namespace (Ex: 'Microsoft.Provider').")
```

**Review by jiasli — APPROVED** [review]
Let's not assume what the author wants to quote. Just merge this PR and let the author change it in the future.

File: src/azure-cli/azure/cli/command_modules/policyinsights/_params.py (hunk 7 above)
Comment (wangzelin007) [reply src/azure-cli/azure/cli/command_modules/policyinsights/_params.py:45]: Should the following three help messages be kept consistent? Personally, I think when parentheses are used, having or not having double quotes is fine.

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**